File: Java/001-twoSum/SolutionByEnum.java
import java.util.Arrays;

/**
 * @Author: Frenude
 * @Description: //TODO
 * @Date: Created in 01 04,2021
 * @Modified By:
 **/
public class SolutionByEnum {

    public static void main(String[] args) {
        int[] nums = {2, 7, 11, 15};
        int target = 9;
        Solution solution = new Solution();
        // lambda 练习
//        BiFunction<int[], Integer, int[]> twoSum = solution::twoSum;
//        int[] ints = twoSum.apply(nums, target);
        int[] ints = solution.twoSum(nums, target);
        System.out.println(Arrays.toString(ints));
    }
}

class Solution {
    public int[] twoSum(int[] nums, int target) {
        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if ((nums[i] + nums[j]) == target) {
                    return new int[]{i, j};
                }
            }
        }
        return new int[0];
    }
}

File: Python/001-twoSum/SolutionByEnum.py
# !/usr/bin/env Python3
# -*- coding: utf-8 -*-

#  @Author: Frenude 
#  @Description:
#  @Date: Created in 01 04,2021
#  @Modified By:
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]


if __name__ == '__main__':
    nums = [2, 7, 11, 15]
    target = 9
    s = Solution()
    print(s.twoSum(nums, target))
